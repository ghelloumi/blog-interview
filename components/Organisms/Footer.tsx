import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.footer`
  margin-top: 3rem;
`;

const Footer: React.FC = () => {
  return (
    <Wrapper>
      <p data-testid="copyright">Â© 2019</p>
    </Wrapper>
  )
}

export default Footer;
