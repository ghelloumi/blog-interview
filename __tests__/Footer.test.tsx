import React from 'react';
import { render } from '@testing-library/react';
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json';
import '@testing-library/jest-dom/extend-expect';

import Footer from '../components/Organisms/Footer';

describe('Enzyme tests', () => {
    test('should render Footer according to snapshot', () => {
        const wrapper = mount(<Footer />);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})

describe('React testing libray tests', () => {
    test('should render Footer with copyright', () => {
        const { getByTestId } = render(<Footer />);
        const copyrights = getByTestId('copyright');
        expect(copyrights.textContent.trim()).toBe('Â© 2019')
    })
})
