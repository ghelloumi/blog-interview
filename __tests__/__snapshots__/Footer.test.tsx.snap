// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Enzyme tests should render Footer according to snapshot 1`] = `
<Footer>
  <Footer__Wrapper>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Footer__Wrapper-sc-1so89p3-0",
            "isStatic": true,
            "lastClassName": "giKWMg",
            "rules": Array [
              "margin-top:3rem;",
            ],
          },
          "displayName": "Footer__Wrapper",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Footer__Wrapper-sc-1so89p3-0",
          "target": "footer",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <footer
        className="Footer__Wrapper-sc-1so89p3-0 giKWMg"
      >
        <p
          data-testid="copyright"
        >
          Â© 2019
        </p>
      </footer>
    </StyledComponent>
  </Footer__Wrapper>
</Footer>
`;
